src/Raven.Server/Documents/Indexes/IndexStorage.cs

```diff
@@ -589,6 +589,35 @@ public void RemoveReferences(Slice key, string collection, HashSet<Slice> refere
             }
         }
 
+        public void RemoveReferencesByPrefix(Slice prefixKey, string collection, HashSet<Slice> referenceKeysToSkip, RavenTransaction tx)
+        {
+            var referencesTree = tx.InnerTransaction.ReadTree(IndexSchema.References);
+
+            using (var it = referencesTree.Iterate(false))
+            {
+                it.SetRequiredPrefix(prefixKey);
+
+                if (it.Seek(prefixKey) == false)
+                    return;
+
+                do
+                {
+                    var key = it.CurrentKey.Clone(tx.InnerTransaction.Allocator);
+
+                    try
+                    {
+                        RemoveReferences(key, collection, referenceKeysToSkip, tx);
```

Comment:
I cannot recall details, but removing from the same tree that we are iterating could cause a corruption.

Reply:
This is fine, we iterate once and re-create the iteration afterward